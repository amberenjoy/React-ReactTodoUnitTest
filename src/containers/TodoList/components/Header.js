import React from 'react';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
  }

  handleInputChange = e => {
    this.setState({
      value: e.target.value
    });
  };

  handleInputKeyUp = e => {
    const { value } = this.state;
    if (e.keyCode === 13 && value) {
      this.props.addUndoItem(value);
      this.setState({ value: '' });
    }
  };

  render() {
    const { value } = this.state;
    return (
      <div className='header'>
        <div className='header-content'>
          TodoList
          <input
            placeholder='Add Todo Item'
            className='header-input'
            data-test='input'
            value={value}
            onChange={this.handleInputChange}
            onKeyUp={this.handleInputKeyUp}
          />
        </div>
      </div>
    );
  }
}

export default Header;
