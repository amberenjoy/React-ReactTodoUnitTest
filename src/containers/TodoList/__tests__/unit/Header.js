import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import { findTestWrapper } from '../../../../utils/testUtils';

describe('Header test', () => {
  it('should render header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a input element', () => {
    const wrapper = shallow(<Header />);
    const inputElem = findTestWrapper(wrapper, 'input');
    expect(inputElem.length).toBe(1);
  });

  it('should set default value of the input element', () => {
    const wrapper = shallow(<Header />);
    const inputElem = findTestWrapper(wrapper, 'input');
    expect(inputElem.prop('value')).toEqual('');
  });

  it('should change input value when user type in5', () => {
    const wrapper = shallow(<Header />);
    const inputElem = findTestWrapper(wrapper, 'input');
    const userInput = 'Study Jest';
    inputElem.simulate('change', {
      target: { value: userInput }
    });
    expect(wrapper.state('value')).toEqual(userInput);
  });

  it('should submit failed, if input null', () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const inputElem = findTestWrapper(wrapper, 'input');
    wrapper.setState({ value: '' });
    inputElem.simulate('keyUp', {
      keyCode: 13
    });
    expect(fn).not.toHaveBeenCalled();
  });

  it('should submit input notNull value and clear input value after submitted', () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const inputElem = findTestWrapper(wrapper, 'input');
    const userInput = 'Study Jest';
    wrapper.setState({ value: userInput });
    inputElem.simulate('keyUp', {
      keyCode: 13
    });
    expect(fn).toHaveBeenCalledWith(userInput);
    const newInputElem = findTestWrapper(wrapper, 'input');
    expect(newInputElem.prop('value')).toBe('');
  });
});
