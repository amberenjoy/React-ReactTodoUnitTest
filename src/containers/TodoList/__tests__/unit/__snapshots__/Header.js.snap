// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should render header 1`] = `ShallowWrapper {}`;
